flink-yarn/src/main/java/org/apache/flink/yarn/entrypoint/YarnSessionClusterEntrypoint.java

```diff
@@ -73,7 +76,17 @@ public static void main(String[] args) {
 			LOG.warn("Could not log YARN environment information.", e);
 		}
 
-		Configuration configuration = YarnEntrypointUtils.loadConfiguration(workingDirectory, env);
+		final CommandLineParser<Configuration> commandLineParser = new CommandLineParser<>(new DynamicParametersConfigurationParserFactory());
+		Configuration dynamicParameters = null;
+		try {
+			dynamicParameters = commandLineParser.parse(args);
+		} catch (FlinkParseException e) {
+			LOG.error("Could not parse command line arguments {}.", args, e);
+			commandLineParser.printHelp(YarnSessionClusterEntrypoint.class.getSimpleName());
+			System.exit(1);
+		}
```

Comment:
Maybe one could introduce

```
private static <T> T parseArgumentsOrExit(
			Logger logger,
			String componentName,
			String[] args,
			ParserResultFactory<T> parserResultFactory) {

		final CommandLineParser<T> commandLineParser = new CommandLineParser<>(parserResultFactory);
		try {
			return commandLineParser.parse(args);
		} catch (FlinkParseException e) {
			logger.error("Could not parse command line arguments {}.", args, e);
			commandLineParser.printHelp(componentName);
			System.exit(1);
			return null;
		}
	}
```

into `ClusterEntrypointUtils` and use it across the cluster entrypoints. This would also ensure that we fail with a consistent error code if the parsing fails.

Reply:
Ok, fair enough. I addressed the comments and pushed the changes. FYI: I also moved `ClusterEntrypointUtils` into the package of `ClusterEntrypoint` (i.e. `org.apache.flink.runtime.entrypoint`) since I wanted to use `ClusterEntrypoint.STARTUP_FAILURE_RETURN_CODE` in the newly introduced utility method.

Alternatively, I could have made `ClusterEntrypoint.STARTUP_FAILURE_RETURN_CODE` public. But moving `ClusterEntrypointUtils` felt like the right thing to do considering that other utility classes (e.g. `ClusterEntrypointExceptionUtils`) are also located in `org.apache.flink.runtime.entrypoint`. Any objections against that?